src/wallet/rpcwallet.cpp

```diff
@@ -3388,6 +3389,368 @@ class CSecp256k1Init {
 static CSecp256k1Init instance_of_csecp256k1;
 }
 
+/* Takes federation redeeem script and adds SHA2(witnessProgram) as a tweak to each pubkey */
```

Comment:
Recommend moving this out of rpcwallet.cpp and making it a general utility function which can also be used in consensus.

Right now this function looks correct but its mirror code (see my comment on https://github.com/ElementsProject/elements/pull/260/commits/65158b7db0445ca541d5e420f75dc0980cb28c62) in peg-in validation is wrong.

BTW is there a test that catches this discrepancy?

Reply:
the updated code seems to match, you're pointing at an old commit perhaps? Agreed on making it a utility function.